File: plugins/ast-tools/tests/fixtures/multi-lang/sample.py
"""Sample Python file for testing AST patterns"""


def greet(name: str) -> str:
    """Greet someone by name."""
    return f"Hello, {name}!"


def farewell(name: str) -> str:
    """Say goodbye to someone."""
    return f"Goodbye, {name}!"


class Greeter:
    """A class that greets people."""

    def __init__(self, greeting: str):
        self.greeting = greeting

    def say(self, name: str) -> str:
        """Say greeting to name."""
        return f"{self.greeting}, {name}!"


# Simple assignments
x = 1
y = "hello"
z = [1, 2, 3]

File: plugins/ast-tools/tests/fixtures/multi-lang/sample.rs
// Sample Rust file for testing AST patterns

/// A struct that greets people
pub struct Greeter {
    greeting: String,
}

impl Greeter {
    /// Create a new greeter
    pub fn new(greeting: &str) -> Self {
        Greeter {
            greeting: greeting.to_string(),
        }
    }

    /// Greet someone by name
    pub fn greet(&self, name: &str) -> String {
        format!("{}, {}!", self.greeting, name)
    }
}

/// Say goodbye to someone
pub fn farewell(name: &str) -> String {
    format!("Goodbye, {}!", name)
}

/// A person struct
pub struct Person {
    pub name: String,
    pub age: u32,
}

fn main() {
    let greeter = Greeter::new("Hello");
    println!("{}", greeter.greet("World"));
}
